test_ffi/tests/test.js

```diff
@@ -465,6 +472,23 @@ uint32Array[0] = 55; // MUTATES!
 console.log("uint32Array[0] after mutation:", uint32Array[0]);
 console.log("Static ptr value after mutation:", view.getUint32());
 
+// Test non-UTF-8 characters
+
+const charView = new Deno.UnsafePointerView(dylib.symbols.static_char);
+
+const charArrayBuffer = charView.getArrayBuffer(14);
+const uint8Array = new Uint8Array(charArrayBuffer);
+assertEquals([...uint8Array], [
+  0xC0, 0xC1, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF,
+  0x00
+]);
+
+try {
+  assertThrows(() => charView.getCString(), TypeError, "Invalid CString pointer, not valid UTF-8");
+} catch (_err) {
+  console.log("Invalid UTF-8 characters to `v8::String`:", charView.getCString());
```

Comment:
This control flow throws me off: if it doesn't throw the *expected* exception, then `assertThrows` throws an `AssertionError`, which is then caught and ignored. By the point that we reach the `catch` block, the string is should be valid (as it wasn't considered invalid UTF-8), but we log that the characters were invalid anyways, and print the valid string?

Reply:
(I cannot find the log "Invalid UTF-8 characters to `v8::String`:" in the checks logs)

Reply:
Ah, yeah I actually implemented this in a bit of an odd way. There are two failure conditions: Either `getCString()` didn't throw or it threw the wrong error.

Now, if `getCString()` threw the wrong error, it will throw the same error again in the catch block which will terminate the program and fail the test case with an assertion on the exit status and all that jazz.

If `getCString()` however did NOT throw, this will log `"Invalid UTF-8 characters to `v8::String`: _____"` with the string. The integration tests do not expect that line to be logged and the test will then fail with the actual output and the expected output being printed and available for comparison to find this line there.

You can get this log to be visible if you revert the first two commits of this PR, or by picking the last two commits on top of current main, and then running the test.

Reply:
> Now, if `getCString()` threw the wrong error, it will throw the same error again in the catch block which will terminate the program and fail the test case with an assertion on the exit status and all that jazz.

Ohh! I was not thinking about that at all. Understood.